website/docs: add docs about entra provider

tanberry opened the pull request:
Draft of docs about how to set up Entra ID and then add it is an provider in authentik. Needs a careful technical review.

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

kensternberg-authentik committed:
web: fix value handling inside controlled components (#9648)

* web: fix esbuild issue with style sheets

Getting ESBuild, Lit, and Storybook to all agree on how to read and parse stylesheets is a serious
pain. This fix better identifies the value types (instances) being passed from various sources in
the repo to the three *different* kinds of style processors we're using (the native one, the
polyfill one, and whatever the heck Storybook does internally).

Falling back to using older CSS instantiating techniques one era at a time seems to do the trick.
It's ugly, but in the face of the aggressive styling we use to avoid Flashes of Unstyled Content
(FLoUC), it's the logic with which we're left.

In standard mode, the following warning appears on the console when running a Flow:

```
Autofocus processing was blocked because a document already has a focused element.
```

In compatibility mode, the following **error** appears on the console when running a Flow:

```
crawler-inject.js:1106 Uncaught TypeError: Failed to execute 'observe' on 'MutationObserver': parameter 1 is not of type 'Node'.
    at initDomMutationObservers (crawler-inject.js:1106:18)
    at crawler-inject.js:1114:24
    at Array.forEach (<anonymous>)
    at initDomMutationObservers (crawler-inject.js:1114:10)
    at crawler-inject.js:1549:1
initDomMutationObservers @ crawler-inject.js:1106
(anonymous) @ crawler-inject.js:1114
initDomMutationObservers @ crawler-inject.js:1114
(anonymous) @ crawler-inject.js:1549
```

Despite this error, nothing seems to be broken and flows work as anticipated.

* web: fix value handling inside controlled components

This is one of those stupid bugs that drive web developers crazy. The basics are straightforward:
when you cause a higher-level component to have a "big enough re-render," for some unknown
definition of "big enough," it will re-render the sub-components. In traditional web interaction,
those components should never be re-rendered while the user is interacting with the form, but in
frameworks where there's dynamic re-arrangement, part or all of the form could get re-rendered at
any mmoment. Since neither the form nor any of its intermediaries is tracking the values as they're
changed, it's up to the components themselves to keep the user's input-- and to be hardened against
property changes coming from the outside world.

So static memoization of the initial value passed in, and aggressively walling off the values the
customer generates from that field, are needed to protect the user's work from any framework's
dynamic DOM management. I remember struggling with this in React; I had hoped Lit was better, but in
this case, not better enough.

The protocol for "is it an ak-data-control" is "it has a `json()` method that returns the data ready
to be sent to the authentik server."  I missed that in one place, so that's on me.

* Eslint had opinions.

* Added comments to explain something.

Commit:
draft Entra docs and fix rac

Commit:
add to sidebar

Commit:
wrap-up

kensternberg-authentik commented on website/docs/providers/entra/add-entra-provider.md:
```diff
@@ -0,0 +1,61 @@
+
+    *   **Slug**: enter the name of the app as you want it to appear in the URL.
+    *   ** Group**: Optionally, enter a group name, of you want this new application to be grouped with other similar apps.
+    *   ** Provider**: Leave this field empty. For certain types of providers (Entra ID, Google Workspace, and SCIM, for example), a paired application is not needed.
```
Inconsistent spacing here.  Is this deliberate?

tanberry replied:
Thanks, not intentional, just fingers moving too fast and giving into muscle memory of adding spaces.

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+
+---
+
+With the Microsoft Entra ID provider, authentik can be the single source of truth for all users and groups. Configuring Entra ID as a provider allows for auto-discovery of user and group accounts, on-going synching of user data such as email address, name, and status, and integrated data mapping of field names and values.
```
Suggestion: "synchronizing" or "synchronization"

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+With the Microsoft Entra ID provider, authentik can be the single source of truth for all users and groups. Configuring Entra ID as a provider allows for auto-discovery of user and group accounts, on-going synching of user data such as email address, name, and status, and integrated data mapping of field names and values.
+
+-   For instructions to configure your Entra ID system to integrate with authentik, refer to [Configure Entra ID](./setup-entra).
+-   For instructions to add Entra ID as a provider in authentik, refer to [Create a Entra ID provider](./add-entra-provider).
```
"For instructions on how to..."  Or
- "For instructions on configuring your Entra ID system to integrate with authentik..."
- "For instructions on adding Entra ID as a provider in authentik..."

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+
+### Discovery
+
+When first creating the provider and setting it up correctly, the provider will run a discovery and query your Entra ID for all users and groups, and attempt to match them with their respective counterparts in authentik. This discovery takes into consideration any **User filtering** options configured in the provider, such as only linking to authentik users in a specific group or excluding servic ...
```
"...correctly" is doing no work here, and implies the user's competence may not be suffcient.  

Suggestion: "When first creating and configuring the provider, authentik will run a discovery process and query your Entra ID..."

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+
+There are two types of sync: a direct sync and a full sync.
+
+The full sync happens when the provider is initially created and when it is saved. The full sync goes through all users and groups matching the **User filtering** options set and will create/update them in Entra ID. In addition to that, the full sync is also run on a schedule every 4 hours.
```
Suggestion for the last sentence: "After the initial sync, authentik will run a full sync every four hours to ensure the consistency [integrity?] of users and groups." (Also: Can this schedule be changed? Where are the dials and knobs for changing it? Can it be disabled? Would it be costly to do this sync across a cloud barrier, since cloud providers charge you for egress?)

tanberry replied:
All great questions, and as I learn the answers, they will be added to the docs!

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+
+The full sync happens when the provider is initially created and when it is saved. The full sync goes through all users and groups matching the **User filtering** options set and will create/update them in Entra ID. In addition to that, the full sync is also run on a schedule every 4 hours.
+
+The direct sync happens when a user or a group is created/updated/deleted in authentik, or a member is added/removed to/from a group. The direct sync will only forward those changes to Entra ID.
```
Suggestion: "A direct sync happens when a user or group is created, updated or deleted in authentik, or when a member [user?] is added to or removed from a group. When one of these events happens, direct sync automatically forwards those changes to Entra ID."

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+
+The direct sync happens when a user or a group is created/updated/deleted in authentik, or a member is added/removed to/from a group. The direct sync will only forward those changes to Entra ID.
+
+During either sync, if a user or group was created in authentik and a matching user/group exists in Entra ID, authentik will link them together as the discovery would do above.
```
"... authentik will automatically link them together."  I think the rest is distracting fluff, asking the reader to recall something above about _process_ rather than being clear about the intent of this sentence, which is the _consequence_.

Commit:
typos

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+
+During either sync, if a user or group was created in authentik and a matching user/group exists in Entra ID, authentik will link them together as the discovery would do above.
+
+If the user has been deleted in Entra ID outside of authentik, authentik will notice this and re-create and re-link the user.
```
Question: The "the" there only exists half the time, because the other half of the time the prior sentence is talking about _group_ creation.  So which user?  Also, "notice" is a human verb; people notice things.  

Unless I'm mis-reading this, the intent is more like: "During a sync, users present in authentik but not in Entra ID will be re-created in Entra ID and re-linked to their identity within authentik."

tanberry replied:
I'm not sure that the user is actually recreated in Entra...  or only in authentik. @BeryJu ?

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+
+If the user has been deleted in Entra ID outside of authentik, authentik will notice this and re-create and re-link the user.
+
+When a property mapping has an invalid expression, it will cause the sync to stop to prevent errors from being spammed.
```
Suggestion: "Note: Invalid property mappings will cause the sync to halt to prevent damage to either authentik or Entra ID."

kensternberg-authentik replied:
Also, when this happens, who gets notified? How? When?

tanberry replied:
hmmm I don;t want to use the general term "damage." I'll try to make phrasing better though.

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+
+When a property mapping has an invalid expression, it will cause the sync to stop to prevent errors from being spammed.
+
+To handle any kind of network interruptions, authentik will detect transient request failures and retry any sync tasks.
```
Suggestion: "interruption" (singular).  

"In the event of a network interruption, authentik will record request failures and reschedule any sync tasks."  (What's the schedule? Can be it changed? Does it have a back-off? What kind? Linear? Exponential? Fibonnaci?)

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+
+There are a couple of considerations in regard to how authentik data is mapped to Entra ID user/group data by default.
+
+-   For users, authentik only saves the full display name, not separate first an family names.
```
Typo: "first _and_ family".

kensternberg-authentik commented on website/docs/providers/entra/index.md:
```diff
@@ -0,0 +1,48 @@
+
+-   For users, authentik only saves the full display name, not separate first an family names.
+
+-   By default, authentik maps a user’s email, a user’s name, and their active status. For groups, the name is synced.
```
Nitpick: "Maps" is a transitive verb.  "Authentik maps [list] ... "_to what_?  Is the word we want here "sync" instead?

kensternberg-authentik commented on website/docs/providers/entra/setup-entra.md:
```diff
@@ -0,0 +1,25 @@
+
+---
+
+The configuration and set up of your Microsft Entra ID environment must be completed before you [add the new provider](./add-entra-provider.md) in authentik.
```
["setup" or "set-up."](https://grammarist.com/spelling/set-up-vs-setup/)

kensternberg-authentik reviewed (commented):
Lots of nitpicks, notes, a few typos, the usual. Can't tell if I'm more crabby than usual, or if you're just exhausted.

Commit:
ken edits

Commit:
more ken edits

Commit:
tweak titles

Commit:
polishing

Commit:
sidebar tweak

dependabot[bot] committed:
core: bump freezegun from 1.5.0 to 1.5.1 (#9693)

Bumps [freezegun](https://github.com/spulec/freezegun) from 1.5.0 to 1.5.1.
- [Release notes](https://github.com/spulec/freezegun/releases)
- [Changelog](https://github.com/spulec/freezegun/blob/master/CHANGELOG)
- [Commits](https://github.com/spulec/freezegun/compare/1.5.0...1.5.1)

---
updated-dependencies:
- dependency-name: freezegun
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
web: bump glob from 10.3.14 to 10.3.15 in /web (#9697)

Bumps [glob](https://github.com/isaacs/node-glob) from 10.3.14 to 10.3.15.
- [Changelog](https://github.com/isaacs/node-glob/blob/main/changelog.md)
- [Commits](https://github.com/isaacs/node-glob/compare/v10.3.14...v10.3.15)

---
updated-dependencies:
- dependency-name: glob
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
web: bump the esbuild group in /web with 2 updates (#9695)

Bumps the esbuild group in /web with 2 updates: [@esbuild/darwin-arm64](https://github.com/evanw/esbuild) and [@esbuild/linux-arm64](https://github.com/evanw/esbuild).


Updates `@esbuild/darwin-arm64` from 0.21.1 to 0.21.2
- [Release notes](https://github.com/evanw/esbuild/releases)
- [Changelog](https://github.com/evanw/esbuild/blob/main/CHANGELOG.md)
- [Commits](https://github.com/evanw/esbuild/compare/v0.21.1...v0.21.2)

Updates `@esbuild/linux-arm64` from 0.21.1 to 0.21.2
- [Release notes](https://github.com/evanw/esbuild/releases)
- [Changelog](https://github.com/evanw/esbuild/blob/main/CHANGELOG.md)
- [Commits](https://github.com/evanw/esbuild/compare/v0.21.1...v0.21.2)

---
updated-dependencies:
- dependency-name: "@esbuild/darwin-arm64"
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: esbuild
- dependency-name: "@esbuild/linux-arm64"
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: esbuild
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
core: bump google-api-python-client from 2.128.0 to 2.129.0 (#9694)

Bumps [google-api-python-client](https://github.com/googleapis/google-api-python-client) from 2.128.0 to 2.129.0.
- [Release notes](https://github.com/googleapis/google-api-python-client/releases)
- [Changelog](https://github.com/googleapis/google-api-python-client/blob/main/CHANGELOG.md)
- [Commits](https://github.com/googleapis/google-api-python-client/compare/v2.128.0...v2.129.0)

---
updated-dependencies:
- dependency-name: google-api-python-client
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
core: bump psycopg from 3.1.18 to 3.1.19 (#9698)

Bumps [psycopg](https://github.com/psycopg/psycopg) from 3.1.18 to 3.1.19.
- [Changelog](https://github.com/psycopg/psycopg/blob/master/docs/news.rst)
- [Commits](https://github.com/psycopg/psycopg/compare/3.1.18...3.1.19)

---
updated-dependencies:
- dependency-name: psycopg
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

authentik-automation[bot] committed:
core, web: update translations (#9672)

Signed-off-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: rissson <18313093+rissson@users.noreply.github.com>

dependabot[bot] committed:
website: bump @types/react from 18.3.1 to 18.3.2 in /website (#9691)

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 18.3.1 to 18.3.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
web: bump chromedriver from 124.0.2 to 124.0.3 in /tests/wdio (#9692)

Bumps [chromedriver](https://github.com/giggio/node-chromedriver) from 124.0.2 to 124.0.3.
- [Commits](https://github.com/giggio/node-chromedriver/compare/124.0.2...124.0.3)

---
updated-dependencies:
- dependency-name: chromedriver
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
web: bump esbuild from 0.21.1 to 0.21.2 in /web (#9696)

transifex-integration[bot] committed:
translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#9706)

Translate django.po in zh-Hans

100% translated source file: 'django.po'
on 'zh-Hans'.

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

transifex-integration[bot] committed:
translate: Updates for file web/xliff/en.xlf in zh-Hans (#9705)

Translate web/xliff/en.xlf in zh-Hans

100% translated source file: 'web/xliff/en.xlf'
on 'zh-Hans'.

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

transifex-integration[bot] committed:
translate: Updates for file web/xliff/en.xlf in zh_CN (#9703)

* Translate web/xliff/en.xlf in zh_CN

100% translated source file: 'web/xliff/en.xlf'
on 'zh_CN'.

* Translate web/xliff/en.xlf in zh_CN

100% translated source file: 'web/xliff/en.xlf'
on 'zh_CN'.

---------

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

transifex-integration[bot] committed:
translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#9702)

Translate locale/en/LC_MESSAGES/django.po in zh_CN

100% translated source file: 'locale/en/LC_MESSAGES/django.po'
on 'zh_CN'.

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

BeryJu committed:
mostly the same changes as google

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
sources/saml: fix FlowPlanner error due to pickle (#9708)

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

dominic-r committed:
website/docs: update traefik to latest version in proxy provider (#9707)

Commit:
fix conflict

Commit:
fight conflicts

Commit:
conflicts

Commit:
conflicts

Commit:
polish

Commit:
tweaks
